### DISCUSSION TITLE
[Feature] Smart Albums

### DISCUSSION THREAD

**Discussion by ibanez97** [immich-app/immich#2903] [Feature Request]
### The feature

I apologize if this has been requested before, I searched for a similar feature and didn't see it.

I think it would be nice to be able to create smart albums based on search requests.  This would allow for the Google Photos feature of being able to automatically share photos of people.  This could extend to just being able to build albums based on any search criteria.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #1673

**Comment by fivestones** [comment]
@bo0tzz I don't think this is a duplicate. #1673 is for creating new albums automatically based on immich recognizing that it's an event or a trip to somewhere or something like that. I expect what the OP there wanted was something like the way the photos app on an iPhone gives you new albums of events like "Last weekend" or "Trip to San Francisco".

This request, on the other hand, is essentially to make a saved search which looks like an album. I could search for certain things and the search results could automatically be made into an auto-updating album which always showed you all the photos which match that search. (Or I guess the option could be there for it just to make a regular album with the results of the search so it would be static.)

I've seen two requests like this, this and another (#962), which you marked as dups of #1673. I think maybe this and #962 are dups, but I don't think either of them are duplicates of #1673.

Any chance you can reopen this? I'd really like this feature too, regardless of if #1673 every comes to fruition or not. And I think programmatically this one would be a lot easier to do than #1673.

**Reply by dutchkimble** [reply]
Agree, this feature would be great to have. I found this app while looking for a solution that the developer stated as his motive for coding this, so it would be great if both my wife and I could take photos of our two kids from our individual devices, which get put automatically into a shared album between us once their faces get recognised. Similarly for their elder brother, our golden retriever. Everything about Immich is spectacular, and this for me would be the final step to convince the wife to stop using the mainstream third-party hosted option.

**Comment by patienttruth** [comment]
This is exactly the use case I came to this future thread looking for.

It would be great to be able to create a "smart album" which uses criteria such as "includes X face". Other criteria such as location would be pretty cool too, but most immediately I'm used to using face as the criteria.

We're moving away from another platform which we use to automatically identify photos with our kids in them and add it to an album. The album was shared to the grandparents so they frequently get updates on the photos of our kids. They use it so much that within a day or two of pulling my photos off the other platform they sent me a message asking where they went. 😁

**Reply by patienttruth** [reply]
On taking a closer look I am pretty sure this is the same thing as #1673 .

The comments in this part of the thread are exactly what I was thinking about.

https://github.com/immich-app/immich/discussions/1673#discussioncomment-7806684
